pkg/clustermesh/clustermesh.go

```diff
@@ -84,6 +84,9 @@ type ClusterMesh struct {
 	// internal implements the common logic to connect to remote clusters.
 	internal internal.ClusterMesh
 
+	usedClusterIDs      map[uint32]struct{}
+	usedClusterIDsMutex lock.Mutex
```

Comment:
Nit. It could make sense to extract this in a separate object, to loosen the circular dependency between `ClusterMesh` and `remoteCluster`.